src/wd.rs

```diff
@@ -373,3 +372,318 @@ impl NewSessionResponse {
         }
     }
 }
+
+/// The builder of [`PrintConfiguration`].
+#[derive(Debug)]
+pub struct PrintConfigurationBuilder {
+    orientation: PrintOrientation,
+    scale: f64,
+    background: PrintBackground,
+    size: PrintSize,
+    margins: PrintMargins,
+    page_ranges: Vec<PrintPageRange>,
+    shrink_to_fit: bool,
+}
+
+impl Default for PrintConfigurationBuilder {
+    fn default() -> Self {
+        Self {
+            orientation: PrintOrientation::default(),
+            scale: 1.0,
+            background: PrintBackground::default(),
+            size: PrintSize::default(),
+            margins: PrintMargins::default(),
+            page_ranges: Vec::default(),
+            shrink_to_fit: true,
+        }
+    }
+}
+
+impl PrintConfigurationBuilder {
+    /// Builds the [`PrintConfiguration`].
+    ///
+    /// Returns None if:
+    ///  - the scale, the margins or the size are infinite, NaN or negative
+    ///  - the margins overflow the size (e.g. margins.left + margins.right >= page.width)
+    pub fn build(self) -> Option<PrintConfiguration> {
```

Comment:
Should we make this return a `Result` instead of an `Option` to indicate _what_ went wrong?

Reply:
Introduced as `PrintConfigurationError`.

I've also  added a new check that prevents the user from using an invalid `PrintSize`.